packages/shared/lib/sdk/sync.ts

```diff
@@ -413,19 +434,27 @@ export class NangoAction {
         });
     }
 
-    public async getToken(): Promise<string | OAuth1Token | BasicApiCredentials | ApiKeyCredentials | AppCredentials> {
+    public async getToken(): Promise<string | OAuth1Token | BasicApiCredentials | ApiKeyCredentials | AppCredentials | AppStoreCredentials> {
         this.exitSyncIfAborted();
         return this.nango.getToken(this.providerConfigKey as string, this.connectionId as string);
     }
 
-    public async getConnection(): Promise<Connection> {
+    public async getConnection(providerConfigKeyOverride?: string, connectionIdOverride?: string): Promise<Connection> {
         this.exitSyncIfAborted();
-        if (!this.memoizedConnection || Date.now() - this.memoizedConnection.timestamp > MEMOIZED_CONNECTION_TTL) {
-            const connection = await this.nango.getConnection(this.providerConfigKey as string, this.connectionId as string);
-            this.memoizedConnection = { connection, timestamp: Date.now() };
+
+        const providerConfigKey = providerConfigKeyOverride || this.providerConfigKey;
+        const connectionId = connectionIdOverride || this.connectionId;
+
+        if (
+            !this.memoizedConnections.get(`${providerConfigKey}${connectionId}`) ||
+            Date.now() - this.memoizedConnections.get(`${providerConfigKey}${connectionId}`)!.timestamp > MEMOIZED_CONNECTION_TTL
+        ) {
+            const connection = await this.nango.getConnection(providerConfigKey as string, connectionId as string);
+            this.memoizedConnections.set(`${providerConfigKey}${connectionId}`, { connection, timestamp: Date.now() });
             return connection;
         }
-        return this.memoizedConnection.connection;
+
+        return this.memoizedConnections.get(`${providerConfigKey}${connectionId}`)!.connection;
```

Comment:
Just a coding style nitpick, it's kinda hard to read, I would store this variable instead of computing it 4 times.

```ts
const couple = `${providerConfigKey}${connectionId}`;
const exists = this.memoizedConnections.get(couple);
if (!exists || Date.now() - exists.timestamp) {
  [...]
  set(couple, { connection });
}

return exists.connection;
```

Reply:
Nice suggestion, updated with https://github.com/NangoHQ/nango/pull/1841/commits/f69f07cdf093d9df8ea8492e69d850dbdc17c6a0

Thanks.